examples/advent2024/day1_part2.carbon

```diff
@@ -0,0 +1,149 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+library "day1_common";
+
+var EOF: i32 = -(1 as i32);
+var push_back: i32 = EOF;
+
+fn ReadCharRaw() -> i32 = "read.char";
```

Comment:
Can these be moved out of `examples/` and into a library?

Reply:
Done.